python/add_alma_barcodes_to_archivesspace.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import json
+from importlib import import_module
+from alma_api_keys import API_KEYS
+from alma_api_client import AlmaAPIClient
+from asnake.client import ASnakeClient
+import asnake.logging as logging
+
+
+logging.setup_logging(filename="archivessnake.log", level="INFO")
+# set label for custom logger - all output will be in archivessnake.log
+logger = logging.get_logger("add_barcodes_to_archivesspace")
+
+
+def get_aspace_containers(aspace_client: ASnakeClient, resource_id: int) -> list[str]:
+    url = f"/repositories/2/resources/{resource_id}/top_containers"
+    container_refs = []
+    for tc in aspace_client.get_paged(url):
+        container_refs.append(tc)
+    # this endpoint returns refs, so we need to get the full container JSON
+    containers = []
+    for tc in container_refs:
+        tc_json = aspace_client.get(tc["ref"]).json()
+        containers.append(tc_json)
+    return containers
+
+
+def get_alma_items(
+    alma_client: AlmaAPIClient, bib_id: str, holdings_id: str
+) -> list[dict]:
+    alma_items = []
+    offset = 0
+    # get the total expected number of items
+    total_items = alma_client.get_items(bib_id, holdings_id, {"limit": 1}).get(
+        "total_record_count"
+    )
+    while len(alma_items) < total_items:
+        current_items = alma_client.get_items(
+            bib_id, holdings_id, {"limit": 100, "offset": offset}
+        )
+        offset += 100
+        alma_items.extend(current_items.get("item"))
+    return alma_items
+
+
+def write_json_to_file(data: list[dict], filename: str, log_message: str) -> None:
```

Comment:
I think it's cleaner to not have this log a message, unless the message can always be built from the other parameters.  For example, if we always wanted to log `filename` as part of the message with other parts constant, like `f"Wrote len{data} items to {filename}"`.  Since the constant text varies, that's not a good fit, and we can't reduce code similarity easily there.

Having the log written in the same method as the file write itself seems good from an atomic POV - if the file write fails, an exception will be thrown here and the log will not be written.  But that's also true if the log message is just written where it's defined.

So, let's not overload `write_json_to_file()` with `log_message`, and just log messages directly where needed instead of passing them around.  This method is still worthwhile for its main purpose.